### DISCUSSION TITLE
upsetjs_r in R shiny, how to align set name to left

### DISCUSSION THREAD

**Discussion by baeywen** [upsetjs/upsetjs#85] [Q&A] [answered]
I am learning upsetjs_r in R shiny. my set name is too long, that is hidden behind the plot. I do not want to change the ration by chartLayout(). is there anyway to make the set name align to the left. The default is to align to the middle.

**Comment by sgratzl** [comment, marked as answer]
there is https://upset.js.org/api/react/interfaces/upsetpropsg.html#setlabelalignment in the library. You can try to set it via `chartProps(setLabelAlignment='left')`

```R
listInput <- list(one = c(1, 2, 3, 5, 7, 8, 11, 12, 13), two = c(1, 2, 4, 5, 10), three = c(1, 5, 6, 7, 8, 9, 10, 12, 13))
upsetjs() %>% fromList(listInput) %>% interactiveChart() %>% chartProps(setLabelAlignment='left')
```

![image](https://user-images.githubusercontent.com/4129778/134666692-93e33339-7061-4088-99b8-7ca8c3279c26.png)

**Reply by baeywen** [reply]
Thank you very much. really helpful
